update dbt-core to v1.7.3

mwhitaker opened the pull request.

mwhitaker commented:
even though dbt-core is now at v1.7.3, some of the adapters can be a bit behind

ARG dbt_core_ref=dbt-core@v1.7.3
ARG dbt_postgres_ref=dbt-core@v1.7.3
ARG dbt_redshift_ref=dbt-redshift@v1.7.1
ARG dbt_bigquery_ref=dbt-bigquery@v1.7.2
ARG dbt_snowflake_ref=dbt-snowflake@v1.7.1
ARG dbt_spark_ref=dbt-spark@v1.7.1

leo-schick commented:
@mwhitaker is there a chance that you can add dbt-databricks as adapter? It is not recommended to use dbt-spark for databricks anymore.

mwhitaker commented:
OK, so this is what I have now:

ARG dbt_core_ref=dbt-core@v1.7.3
ARG dbt_postgres_ref=dbt-core@v1.7.3
ARG dbt_redshift_ref=dbt-redshift@v1.7.1
ARG dbt_bigquery_ref=dbt-bigquery@v1.7.2
ARG dbt_snowflake_ref=dbt-snowflake@v1.7.1
ARG dbt_spark_ref=dbt-spark@v1.7.1
ARG dbt_databricks_ref=dbt-databricks@v1.7.2

Since I don't personally have access to databricks, is there a way for you to test?

leo-schick commented:
@mwhitaker I tested it in a private repository and it works there. Can you upgrade to dbt-databricks@v.1.7.3?

In addition, it would be great to add an additional substitution of environment variable `DBT_HTTP_PATH` to the already existing `DBT_TOKEN`. Then one can create a cluster on the fly, run it and drop it again. Otherwise one has always to use a predefined cluster.

mwhitaker commented:
I updated the databricks adapter to v.1.7.3 and I added support for a new optional `http_path` input. See [here](https://github.com/mwhitaker/dbt-action/blob/f7d10c7aefc32049f086a1c432b5497dec9f5f62/README.md?plain=1#L121)

leo-schick commented:
the optional `_http_path_` seems to works, but the parameter `_token_` does not work. I think you write the databricks token to the wrong file. See here: https://github.com/mwhitaker/dbt-action/blob/master/entrypoint.sh#L30 Should it be not be written to `./profiles.yml` instead of `./datab.yml`?

leo-schick commented on entrypoint.sh:
```diff
@@ -28,6 +28,10 @@ elif [ -n "${DBT_TOKEN}" ]
 then
  echo trying to use DBT_TOKEN/databricks
  sed -i "s/_token_/${DBT_TOKEN}/g" ./datab.yml
+elif [ -n "${INPUT_HTTP_PATH}" ]
```
This seems to be wrong: For Databricks, we need to use both, the token and the http path.

leo-schick commented:
I fixed these issues and others I found in my branch. Check out the commits in [my branch](https://github.com/leo-schick/dbt-action/commits/master/)

mwhitaker commented:
Thank you Leo, I appreciate it. I basically only use this action with BigQuery so I don't have a way to test the other connectors.

I added those changes to this branch. Can you do one more check please?

leo-schick commented:
It works fine for me in my branch. Since I guess this is a fast forward merge, will work here, too. There aren't any differences between my and your branch.
